src/test/java/com/exasol/adapter/dialects/db2/DB2SqlDialectIT.java

```diff
@@ -0,0 +1,233 @@
+        assertVirtualTableContents(table,
+                table().row(-123.45700073242188).row(1.2450000212993473E-4).matches(NO_JAVA_TYPE_CHECK));
+    }
+
+    @Test
+    void testDoubleMapping() throws SQLException {
+        final String table = createSingleColumnTable("DOUBLE", List.of("-1000", "14.568001"));
+        assertVirtualTableContents(table, table().row(-1000).row(14.568001).matches(NO_JAVA_TYPE_CHECK));
+    }
+
+    @Test
+    void testCharMapping() throws SQLException {
+        final String table = createSingleColumnTable("CHAR(15)", List.of("'This is a test'", "'Hello'"), "CHAR");
+        assertVirtualTableContents(table, table().row("This is a test ").row("Hello          ").matches());
+    }
+
+    @Test
+    void testVarcharMapping() throws SQLException {
+        final String table = createSingleColumnTable("VARCHAR(32672)", List.of("'This is a test'", "'Hello'"),
+                "VARCHAR");
+        assertVirtualTableContents(table, table().row("This is a test").row("Hello").matches());
+    }
+
+    @Test
+    void testDateMapping() throws SQLException {
+        final String table = createSingleColumnTable("DATE", List.of("'2018-10-27'", "'2021-12-12'"));
+        assertVirtualTableContents(table,
+                table().row(Date.valueOf("2018-10-27")).row(Date.valueOf("2021-12-12")).matches());
+    }
+
+    @Test
+    void testTimeMapping() throws SQLException {
+        final String table = createSingleColumnTable("TIME", List.of("'00.00.00'", "'23.59.59'"));
+        assertVirtualTableContents(table, table().row("00.00.00").row("23.59.59").matches());
+    }
+
+    @Test
+    void testTimestampMapping() throws SQLException {
+        final String table = createSingleColumnTable("TIMESTAMP",
+                List.of("'1900-01-01 00.00.00'", "'9999-12-31 23.59.59'"));
+        assertVirtualTableContents(table,
+                table().row("1900-01-01-00.00.00.000000").row("9999-12-31-23.59.59.000000").matches());
+    }
+
+    @Test
+    void testXMLMapping() throws SQLException {
+        final String table = createSingleColumnTable("XML",
+                List.of("'<?xml version=\"1.0\" encoding=\"UTF-8\"?><note><heading>Test</heading></note>'"));
+        assertVirtualTableContents(table, table()
+                .row("<?xml version=\"1.0\" encoding=\"UTF-8\"?><note><heading>Test</heading></note>").matches());
+    }
+
+    @Test
+    void testBooleanMapping() throws SQLException {
+        final String table = createSingleColumnTable("BOOLEAN", List.of("TRUE", "FALSE"));
+        assertVirtualTableContents(table, table().row(true).row(false).matches());
+    }
+
+    private String createSingleColumnTable(final String sourceType, final List<String> values) throws SQLException {
+        return createSingleColumnTable(sourceType, values, sourceType);
+    }
+
+    private String createSingleColumnTable(final String sourceType, final List<String> values, final String suffix)
```

Comment:
I think as on the long term we should implement a dialect for all SQL virtual schemas in test-db-builder-java.

Reply:
I understand that you daon't want to do it in  this PR. But can you open a ticket  on test-db-builder and here one for refacvtoring?

Reply:
Yes, it would be nice to have a dialect for it. I'll a ticket